hie-hare/Haskell/Ide/HaRePlugin.hs

```diff
@@ -180,10 +147,28 @@ makeRefactorResult changedFiles = do
 
 -- ---------------------------------------------------------------------
 
-catchException :: (IO t) -> IO (Either String t)
-catchException f = do
-  res <- handle handler (f >>= \r -> return $ Right r)
-  return res
-  where
-    handler:: SomeException -> IO (Either String t)
-    handler e = return (Left (show e))
+
+runHareCommand :: T.Text -- ^ The file name we'll operate on
+                 -> String -- ^ command name for log
+                 -> (RefactSettings -> GM.Options -> FilePath -> IO [FilePath])
+                 -> IdeM (IdeResponse RefactorResult)
+runHareCommand fp name cmd = do
```

Comment:
I wonder if we shouldn't pull this into `PluginUtils` or `MonadFunctions`, to run a `GhcModT` action in an appropriate directory.